gaiku-common/src/data/chunk.rs

```diff
@@ -70,12 +70,16 @@ impl Chunk {
     }
 
     pub fn set(&mut self, x: usize, y: usize, z: usize, value: u8) {
-        let index = self.index(x, y, z);
-        self.values[index] = value;
+        if value != 0 {
+            self.set_with_color(x, y, z, value, [0, 0, 0, 1].into());
```

Comment:
I would check if the color is "empty" or not before set the default one (white), I think something like this should work (didn't tested)

```suggestion
          let color: Vector4<u8> = if self.get_color(x, y, z).unwrap_or([0, 0, 0, 0].into())[3] == 0 {
                   [1, 1, 1, 1].into()
          } else {
                   self.get_color(x, y, z).unwrap()
          }
          self.set_with_color(x, y, z, value,  color);
```

Reply:
Shouldn't self.get_color(x, y, z) always suceed though as long as xyz are in bounds? Since it initializes the vector completely at init.

Reply:
Yes, that's why I suggested the change in the get_color.